src/Provider/AbstractProvider.php

```diff
@@ -206,12 +206,34 @@ public function getAccessToken($grant = 'authorization_code', $params = [])
             // @codeCoverageIgnoreEnd
         }
 
+        $result = $this->parseResponse($response);
+
+        // @codeCoverageIgnoreStart
+        $this->errorCheck($result);
+        // @codeCoverageIgnoreEnd
+
+        $result = $this->prepareAccessTokenResult($result);
+
+        return $grant->handleResponse($result);
+    }
+
+
+    /**
+     * Parse the response, according to the provider response type.
+     *
+     * @param  string $response
+     * @return array
+     */
+    protected function parseResponse($response)
+    {
+        $result = [];
+
         switch ($this->responseType) {
             case 'json':
-                $result = json_decode($response, true);
+                $json = json_decode($response, true);
 
-                if (JSON_ERROR_NONE !== json_last_error()) {
-                    $result = [];
+                if (JSON_ERROR_NONE === json_last_error()) {
+                    $result = $json;
```

Comment:
If there is a JSON error, is it okay for `parseResponse()` to return an empty array? That's what it looks like it would do. Should it instead throw an exception, since it couldn't parse the response?